tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java

```diff
@@ -375,6 +375,32 @@ private void setBreakpoints(List<BallerinaTestDebugPoint> breakPoints) throws Ba
         }
     }
 
+    /**
+     * Can be used to fetch stack frames when a debug hit is occurred.
+     *
+     * @param args debug stopped event arguments.
+     * @return StackFrame array with stack frames information.
+     * @throws BallerinaTestException if an error occurs when fetching stack frames.
+     */
+    protected StackFrame[] fetchStackFrames(StoppedEventArguments args) throws BallerinaTestException {
+        if (!DebugHitListener.connector.isConnected()) {
+            return new StackFrame[0];
```

Comment:
IMO better to throw an `BallerinaTestException`, instead of returning an empty array. (To avoid potential `IndexOutOfBoudException`s in test cases.)

Reply:
Done